index.js

```diff
@@ -26,28 +26,33 @@ function getSortOrder(options) {
         return {};
     }
 
-    // Add sorting indexes to order
-    var order = {};
-
-    if (typeof sortOrder[0] === 'string') {
-        sortOrder.forEach(function (prop, propIndex) {
-            order[prop] = {
-                group: 0,
-                prop: propIndex
-            };
-        });
-    } else {
-        sortOrder.forEach(function (group, groupIndex) {
-            group.forEach(function (prop, propIndex) {
-                order[prop] = {
+    // Create and return sorting indexes
+    return (typeof sortOrder[0] === 'string' ? [sortOrder] : sortOrder)
+        .reduce(function (order, group, groupIndex) {
+            return group.reduce(function (innerOrder, prop, propIndex) {
+                innerOrder[prop] = {
                     group: groupIndex,
                     prop: propIndex
                 };
-            });
-        });
+
+                return innerOrder;
+            }, order);
+        }, {});
+}
+
```

Comment:
I'll better stick to my code in this part. This `reduce` thing and closures are very hard to understand, and the code becomes less maintainable and straightforward.